Formula/gmp.rb

```diff
@@ -14,14 +14,36 @@ class Gmp < Formula
 
   uses_from_macos "m4" => :build
 
+  patch do
+    # Remove when upstream fix is released
+    # https://gmplib.org/list-archives/gmp-bugs/2020-July/004837.html
+    # arm64-darwin patch
+    url "https://gmplib.org/list-archives/gmp-bugs/attachments/20200703/6c9b827c/attachment.bin"
+    sha256 "517ef7c22102e7ce15e71b75e4e4edcd2149dccfcf02a2b2f19f1407107fde18"
+  end
+
+  # Remove when upstream fix is released
+  # Fixes arm64-darwin assembly/linkage issue
+  patch :DATA
+
+  depends_on "autoconf" if Hardware::CPU.arm?
+  depends_on "automake" if Hardware::CPU.arm?
+  depends_on "libtool" if Hardware::CPU.arm?
+
   def install
-    # Work around macOS Catalina / Xcode 11 code generation bug
-    # (test failure t-toom53, due to wrong code in mpn/toom53_mul.o)
-    ENV.append_to_cflags "-fno-stack-check"
+    if Hardware::CPU.arm?
+      args =  %W[--prefix=#{prefix}]
+      args << "--build=aarch64-apple-darwin#{`uname -r`.to_i}"
+      system "autoreconf", "-fiv"
+    else
+      # Work around macOS Catalina / Xcode 11 code generation bug
+      # (test failure t-toom53, due to wrong code in mpn/toom53_mul.o)
+      ENV.append_to_cflags "-fno-stack-check"
 
-    # Enable --with-pic to avoid linking issues with the static library
-    args = %W[--prefix=#{prefix} --enable-cxx --with-pic]
-    args << "--build=#{Hardware.oldest_cpu}-apple-darwin#{`uname -r`.to_i}"
+      # Enable --with-pic to avoid linking issues with the static library
+      args = %W[--prefix=#{prefix} --enable-cxx --with-pic]
```

Comment:
Why are `--enable-cxx --with-pic` not passed on ARM? These should not be platform-dependent.

Reply:
@fxcoudert passing these caused failures for me when producing a generic C build and before adding the patch. Can re-test this soon to confirm whether or not this is still the case.

Reply:
Confirmed this works since including the patch